warehouse/static/js/warehouse/controllers/horizontal_tabs_controller.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static targets = ["tab", "tabPanel"];
+
+  connect() {
+  }
+
+  initialize() {
+    this.showTab();
+  }
+
+  change(e) {
+    this.index = this.tabTargets.indexOf(e.target);
+    this.showTab(this.index);
+  }
+
+  showTab() {
+    this.tabPanelTargets.forEach((el, i) => {
+      if (i == this.index) {
+        el.classList.remove("is-hidden");
+      } else {
+        el.classList.add("is-hidden");
+      }
+    });
+
+    this.tabTargets.forEach((el, i) => {
+      if (i == this.index) {
+        el.classList.add("is-active");
+      } else {
+        el.classList.remove("is-active");
+      }
+    });
+  }
+
+  get index() {
+    return parseInt(this.data.get("index"));
+  }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏this approach uses an index - where we're relying on the correct ordering of two elements to be always correct (which if we do it right, they should be).

One common way is to use matching `id` values on the buttons and the content, and thus match them up, like so:

```html
<button id="github" ...>
...
<div id="github" class="tabcontent ... >
```

And then you can match lookups by ID instead of index, and activate whichever one the user clicks - and not have to mutate the index value in the HTML.

Reply:
My understanding is that element IDs should be unique within the DOM, so I'm not sure this approach would be valid without adding some kind of predictable prefix/suffix to the IDs.

I think the risk of having buttons/tabs out of order is probably small, so I'm inclined to leave this as-is.